### PR TITLE
s3mapprovider for Collector: load configuration from config files in Amazon S3

### PR DESCRIPTION
**Pull request by rice-junhaoyu**
**Description:** <Describe what has changed.>
An implementation of ConfigMapProvider for Amazon S3 (s3mapprovider) allows OTEL Collector the ability to load configuration for itself by fetching and reading config files stored in Amazon S3.

This PR is the same PR as #13113, but requires me to open a new one due to my deactivated email address. It was already reviewed by @mx-psi and @Aneurysm9.

**Link to tracking Issue:** <Issue number if applicable>
#12939

**Testing:** <Describe what testing was performed and which tests were added.>
First, I mocked how a S3 client works while interacting with a S3 server.
Second, I designed and passed all the unit tests for edge cases such as empty fields, invalid config files, non-exist config files and so on.
Third, I did integration tests such as:
- Start an OTEL Collector with s3mapprovider.
- Start an OTEL sample application to send telemetry data to check if the OTEL Collector is running well.

**Documentation:** <Describe the documentation added.>
Added a README.md

**Sponsor:**
@Aneurysm9

### CODE REVIEW

File: .github/ALLOWLIST
Comment (mx-psi) [inline .github/ALLOWLIST:17]:
```suggestion
confmap/provider/s3mapprovider
```

**Review by mx-psi — APPROVED** [review]
LGTM, I will merge once the CI passes. Thanks @rice-junhaoyu for taking the time to open this new PR and address the required changes :)

**Comment by jufemaiz** [comment]
Is this implementation operational in v0.76.0? I'm getting the following:

```
collector server run finished with error: unsupported scheme on URI "s3://BUCKET.s3.REGION.amazonaws.com/KEY"
```

**Comment by mx-psi** [comment]
This is available since v0.61.0. v0.76.0 has not yet been released :) If you are having trouble please open an issue so that the codeowners can help you

**Comment by Aneurysm9** [comment]
This is not available in the collector-contrib distribution as the collector builder does not yet have support for specifying which config providers to include.  See https://github.com/open-telemetry/opentelemetry-collector/issues/4759.

**Comment by jufemaiz** [comment]
@Aneurysm9 oh thank you for that clarification! It's not clear from the [docs](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/14317/files#diff-acf7b18fdaf01cae270292cca2fac09f721888befae881c3bdc8635885ac1171) that that was the case!

**Pull request merged**
